JetBrains: Crash occurred while the autocomplete is triggered

### IDE Information

PyCharm 2023.3.3 (Community Edition)
Build #PC-233.13763.11, built on January 26, 2024
Runtime version: 17.0.9+7-b1087.11 aarch64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
macOS 14.3
GC: G1 Young Generation, G1 Old Generation
Memory: 2048M
Cores: 8
Metal Rendering is ON
Non-Bundled Plugins:
  com.sourcegraph.jetbrains (6.0.23-nightly)


### Bug Description

Platform: macOS

**Steps to reproduce:**

1. Have Cody plugin installed in your JB IntelliJ editor
2. Login to Cody with Free/Pro user
3. Relaunch the IDE and start typing in a source code file and verify that autocompletion triggered as you type

### Expected behavior:
Crash should not occur.

### Observed behavior:
Crash occurred when the autocompletion triggered with following stacktrace.
Stacktrace:
```text
java.lang.AssertionError: Start: 0, end: 0, length: 0
	at com.intellij.openapi.editor.impl.ComplementaryFontsRegistry.getFontAbleToDisplay(ComplementaryFontsRegistry.java:125)
	at com.sourcegraph.cody.autocomplete.render.CodyAutocompleteElementRenderer.fontInfoForText(CodyAutocompleteElementRenderer.kt:43)
	at com.sourcegraph.cody.autocomplete.render.CodyAutocompleteBlockElementRenderer.paint(CodyAutocompleteBlockElementRenderer.kt:37)
	at com.intellij.openapi.editor.EditorCustomElementRenderer.paint(EditorCustomElementRenderer.java:76)
	at com.intellij.openapi.editor.impl.view.EditorPainter$Session.paintBlockInlays(EditorPainter.java:1231)
	at com.intellij.openapi.editor.impl.view.EditorPainter$Session.paint(EditorPainter.java:202)
	at com.intellij.openapi.editor.impl.view.EditorPainter.paint(EditorPainter.java:72)
	at com.intellij.openapi.editor.impl.view.EditorView.paint(EditorView.java:282)
	at com.intellij.openapi.editor.impl.EditorImpl.lambda$paint$48(EditorImpl.java:2007)
	at com.intellij.openapi.application.ReadAction.lambda$run$1(ReadAction.java:53)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:909)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:65)
	at com.intellij.openapi.application.ReadAction.run(ReadAction.java:52)
	at com.intellij.openapi.editor.impl.EditorImpl.paint(EditorImpl.java:1989)
	at com.intellij.openapi.editor.impl.EditorComponentImpl.lambda$paintComponent$1(EditorComponentImpl.java:277)
	at com.intellij.openapi.application.ReadAction.lambda$run$1(ReadAction.java:53)
	at com.intellij.openapi.application.impl.ApplicationImpl.runReadAction(ApplicationImpl.java:909)
	at com.intellij.openapi.application.ReadAction.compute(ReadAction.java:65)
	at com.intellij.openapi.application.ReadAction.run(ReadAction.java:52)
	at com.intellij.openapi.editor.impl.EditorComponentImpl.paintComponent(EditorComponentImpl.java:265)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1115)
	at com.intellij.openapi.editor.impl.EditorComponentImpl.paint(EditorComponentImpl.java:143)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1124)
	at java.desktop/javax.swing.JViewport.paint(JViewport.java:736)
	at com.intellij.ui.components.JBViewport.paint(JBViewport.java:235)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1124)
	at java.desktop/javax.swing.JComponent.paintChildren(JComponent.java:955)
	at java.desktop/javax.swing.JComponent.paint(JComponent.java:1124)
	at java.desktop/javax.swing.JLayeredPane.paint(JLayeredPane.java:586)
	at java.desktop/javax.swing.JComponent.paintToOffscreen(JComponent.java:5312)
	at java.desktop/javax.swing.BufferStrategyPaintManager.paint(BufferStrategyPaintManager.java:247)
	at java.desktop/javax.swing.RepaintManager.paint(RepaintManager.java:1347)
	at java.desktop/javax.swing.JComponent._paintImmediately(JComponent.java:5260)
	at java.desktop/javax.swing.JComponent.paintImmediately(JComponent.java:5070)
	at java.desktop/javax.swing.RepaintManager$4.run(RepaintManager.java:882)
	at java.desktop/javax.swing.RepaintManager$4.run(RepaintManager.java:865)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:865)
	at java.desktop/javax.swing.RepaintManager.paintDirtyRegions(RepaintManager.java:838)
	at java.desktop/javax.swing.RepaintManager.prePaintDirtyRegions(RepaintManager.java:787)
	at java.desktop/javax.swing.RepaintManager$ProcessingRunnable.run(RepaintManager.java:1909)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:792)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:739)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:733)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:761)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:695)
	at com.intellij.ide.IdeEventQueue._dispatchEvent$lambda$12(IdeEventQueue.kt:589)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithoutImplicitRead(RwLockHolder.kt:44)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:589)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:72)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:355)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:354)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:793)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:354)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:349)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:1014)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:106)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:1014)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$7(IdeEventQueue.kt:349)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:386)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)

```

### Additional context

**Video link:**

https://drive.google.com/file/d/1S_bwEw-58PqU3niNwea4xOsT1q-TaFYQ/view?usp=drive_link

--- a/src/main/kotlin/com/sourcegraph/cody/autocomplete/CodyAutocompleteManager.kt
+++ b/src/main/kotlin/com/sourcegraph/cody/autocomplete/CodyAutocompleteManager.kt
@@ -424,27 +424,30 @@ class CodyAutocompleteManager {
 
     var inlay: Inlay<*>? = null
     if (startsInline) {
-      val renderer =
-          CodyAutocompleteSingleLineRenderer(
-              completionText.lines().first(), items, editor, AutocompleteRendererType.INLINE)
-      inlay =
-          inlayModel.addInlineElement(cursorOffset, /* relatesToPrecedingText = */ true, renderer)
+      val text = completionText.lines().first()
+      if (text.isNotEmpty()) {
+        val renderer =
+            CodyAutocompleteSingleLineRenderer(text, items, editor, AutocompleteRendererType.INLINE)
+        inlay =
+            inlayModel.addInlineElement(cursorOffset, /* relatesToPrecedingText = */ true, renderer)
+      }
     }
     val lines = completionText.lines()
     if (lines.size > 1) {
       val text =
           (if (startsInline) lines.drop(1) else lines).dropWhile { it.isBlank() }.joinToString("\n")
-      if (text.isEmpty()) return
-      val renderer = CodyAutocompleteBlockElementRenderer(text, items, editor)
-      val inlay2 =
-          inlayModel.addBlockElement(
-              /* offset = */ cursorOffset,
-              /* relatesToPrecedingText = */ true,
-              /* showAbove = */ false,
-              /* priority = */ Int.MAX_VALUE,
-              /* renderer = */ renderer)
-      if (inlay == null) {
-        inlay = inlay2
+      if (text.isNotEmpty()) {
+        val renderer = CodyAutocompleteBlockElementRenderer(text, items, editor)
+        val inlay2 =
+            inlayModel.addBlockElement(
+                /* offset = */ cursorOffset,
+                /* relatesToPrecedingText = */ true,
+                /* showAbove = */ false,
+                /* priority = */ Int.MAX_VALUE,
+                /* renderer = */ renderer)
+        if (inlay == null) {
+          inlay = inlay2
+        }
       }
     }
 

--- a/src/main/kotlin/com/sourcegraph/cody/autocomplete/CodyAutocompleteManager.kt
+++ b/src/main/kotlin/com/sourcegraph/cody/autocomplete/CodyAutocompleteManager.kt
@@ -434,6 +434,7 @@ class CodyAutocompleteManager {
     if (lines.size > 1) {
       val text =
           (if (startsInline) lines.drop(1) else lines).dropWhile { it.isBlank() }.joinToString("\n")
+      if (text.isEmpty()) return
       val renderer = CodyAutocompleteBlockElementRenderer(text, items, editor)
       val inlay2 =
           inlayModel.addBlockElement(

